torch/distributed/rpc/api.py

```diff
@@ -640,26 +621,7 @@ def _invoke_rpc(to, func, rpc_type, args=None, kwargs=None, rpc_timeout=UNSET_RP
 
     ctx_manager = contextlib.suppress()
     if should_profile:
-        # Create appropriate string representation based on type of func
-        # (builtin, script, python)
-        if qualified_name is None:
-            func_name = (
-                torch._jit_internal._qualified_name(func)
-                if isinstance(func, torch.jit.ScriptFunction)
-                else func.__qualname__
-            )
-        else:
-            func_name = qualified_name
-        # Build RPC profiling key.
-        rpc_profiling_key = _build_rpc_profiling_key(
-            rpc_type,
-            func_name,
-            get_worker_info().name,
-            dst_worker_info.name,
-        )
-        RemoteProfilerManager.set_current_profiling_key(rpc_profiling_key)
-        # Mypy doesn't support re-def of a variable not in the same block (#1174)
-        ctx_manager = torch.autograd.profiler.record_function(rpc_profiling_key)  # type: ignore[assignment]
+        ctx_manager = _enable_rpc_profiler(qualified_name, func, rpc_type, dst_worker_info)
```

Comment:
Could you also include the logic for `should_profile` and contextlib.suppress() to the helper function (i.e. the code starting on 620)? Also, feel free to remove the TODO above